pkg/util/util.go

```diff
@@ -1030,6 +1030,10 @@ func DownloadFileInMemory(url string) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
+	// We have a non 1xx / 2xx status, return an error
+	if (resp.StatusCode - 300) > 0 {
+		return nil, errors.Errorf("fail to retrive %s: %s", url, http.StatusText(resp.StatusCode))
+	}
```

Comment:
trying to understand, could you explain why we do StatusCode - 300 here?

probably also mention the word status code in the err msg

Reply:
This is from https://github.com/devfile/library/blob/94ab5db4e840180c1e81a65214aee287c0bb4a8c/pkg/util/util.go#L769-L772
Because status >= 300 means with error

Reply:
updated